### PR TITLE
Added dark mode

### PR DESCRIPTION
**Pull request by Dhravya**
This PR adds darkMode.

For storing the theme between sessions, I've made use of cookies:
```ts
const theme = cookies().get("theme")
````

I've just use the colors and icons to the best of my knowledge. Please feel free to change them as you like 😄

Here's a demo
Demo

### CODE REVIEW

File: app/layout.tsx [hunk 1]
```diff
@@ -7,6 +7,7 @@ import { Analytics } from "@vercel/analytics/react";
 import { Metadata } from "next";
 import Toaster from "./toaster";
 import { ReactNode } from "react";
+import { cookies } from "next/dist/client/components/headers";
```
Comment (steven-tey) [inline app/layout.tsx:10]: This should be `import { cookies } from 'next/headers'`

**Comment by Dhravya** [comment]
@steven-tey I have made the changes :)

**Comment by Dhravya** [comment]
As a twitter comment mentioned, it was a bit too dark for the eyes. https://twitter.com/_Eduard26/status/1674449469193039873?s=20 

I tried to soften it
Here, have a look

**Comment by Dhravya** [comment]
Just realised that checkmarks weren't styled, fixed

checkmark fix

**Comment by Dhravya** [comment]
The slash and color selector menu were completely black earlier which didn't really look good. I fixed that and 

Also fixed the magic icon not appearing correctly
image

**Comment by Dhravya** [comment]
Hey @steven-tey let me know if there are any more improvements!

**Comment by steven-tey** [comment]
I made a switch to using CSS variables instead so we don't need to repeatedly do `dark:` for everything – just `*-stone-*` will do! <picture data-single-emoji=":yayawesome:" title=":yayawesome:">:yayawesome:</picture> 

CleanShot 2023-07-04 at 23 48 04

**Review by steven-tey — APPROVED** [review]

**Pull request merged**